File: tests/fixtures/indexer_inputs/typescript_repo/src/providers/anthropic.ts
import { Provider } from "./base";

export class AnthropicProvider extends Provider {
  private apiKey: string;

  constructor(apiKey: string) {
    super();
    this.apiKey = apiKey;
  }

  async complete(prompt: string): Promise<string> {
    return this.call(prompt);
  }

  private call(prompt: string): Promise<string> {
    return Promise.resolve(prompt);
  }
}

File: tests/fixtures/indexer_inputs/python_repo/app/providers/anthropic.py
from .base import Provider


class AnthropicProvider(Provider):
    def __init__(self, api_key: str) -> None:
        self.api_key = api_key

    def complete(self, prompt: str) -> str:
        return self._call(prompt)

    def _call(self, prompt: str) -> str:
        return prompt
